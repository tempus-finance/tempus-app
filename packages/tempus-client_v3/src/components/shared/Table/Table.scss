$tableColumnPadding: 8px;
$tableSidePadding: 20px;
$tableRowHeight: 80px;

.tc__table {
  width: 100%;
  border-spacing: 0px 20px;
}
