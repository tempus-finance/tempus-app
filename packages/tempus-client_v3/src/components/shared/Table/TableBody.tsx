import { FC } from 'react';

const TableBody: FC = props => {
  const { children } = props;

  return <tbody>{children}</tbody>;
};
export default TableBody;
