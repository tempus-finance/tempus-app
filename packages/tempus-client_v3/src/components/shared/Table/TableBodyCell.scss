.tc__tableBodyCell {
  border-width: 1px 0px;
  border-color: #bfc0c0;
  border-style: solid;
  height: 80px;
  padding-right: 8px;

  &:first-child {
    border-left-width: 1px;
    border-color: #bfc0c0;
    border-radius: 40px 0px 0px 40px;
    padding-left: 20px;
  }

  &:last-child {
    border-right-width: 1px;
    border-color: #bfc0c0;
    border-radius: 0px 40px 40px 0px;
    padding-right: 20px;
  }

  &[data-align='left'] {
    text-align: left;
  }

  &[data-align='center'] {
    text-align: center;
  }

  &[data-align='right'] {
    text-align: right;
  }
}
