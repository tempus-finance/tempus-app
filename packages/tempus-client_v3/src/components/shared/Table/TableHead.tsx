import { FC } from 'react';

const TableHead: FC = props => {
  const { children } = props;

  return <thead>{children}</thead>;
};
export default TableHead;
