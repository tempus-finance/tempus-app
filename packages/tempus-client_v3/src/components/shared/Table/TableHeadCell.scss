@use './Table.scss' as TableStyle;

.tc__tableHeadCell {
  padding-right: TableStyle.$tableColumnPadding;

  &:first-child {
    padding-left: TableStyle.$tableSidePadding;
  }

  &:last-child {
    padding-right: TableStyle.$tableSidePadding;
  }

  &[data-align='left'] {
    text-align: left;
  }

  &[data-align='center'] {
    text-align: center;
  }

  &[data-align='right'] {
    text-align: right;
  }
}
