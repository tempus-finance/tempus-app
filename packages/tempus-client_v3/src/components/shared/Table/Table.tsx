import { FC } from 'react';

import './Table.scss';

const Table: FC = props => {
  const { children } = props;

  return <table className="tc__table">{children}</table>;
};
export default Table;
