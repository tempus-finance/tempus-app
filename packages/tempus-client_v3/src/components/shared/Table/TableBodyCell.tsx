import { FC } from 'react';

import './TableBodyCell.scss';

interface TableBodyCellProps {
  align?: 'left' | 'center' | 'right';
}

const TableBodyCell: FC<TableBodyCellProps> = props => {
  const { align = 'left', children } = props;

  return (
    <td className="tc__tableBodyCell" data-align={align}>
      {children}
    </td>
  );
};
export default TableBodyCell;
