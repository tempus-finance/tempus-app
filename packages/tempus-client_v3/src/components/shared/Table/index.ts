import Table from './Table';
import TableHead from './TableHead';
import TableHeadCell from './TableHeadCell';
import TableBody from './TableBody';
import TableBodyCell from './TableBodyCell';
import TableRow from './TableRow';

export { Table, TableHead, TableHeadCell, TableBody, TableBodyCell, TableRow };
