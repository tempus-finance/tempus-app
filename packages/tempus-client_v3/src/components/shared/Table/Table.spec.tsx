import { fireEvent, render } from '@testing-library/react';
import Table from './Table';
import TableBody from './TableBody';
import TableBodyCell from './TableBodyCell';
import TableHead from './TableHead';
import TableHeadCell from './TableHeadCell';
import TableRow from './TableRow';

const mockOnClickHandler = jest.fn();

const subject = () =>
  render(
    <Table>
      <TableHead>
        <TableRow id="header">
          <TableHeadCell label="Asset" align="right" />
          <TableHeadCell label="Protocol" />
        </TableRow>
      </TableHead>
      <TableBody>
        <TableRow id="first-pool" onClick={mockOnClickHandler}>
          <TableBodyCell align="center">ETH</TableBodyCell>
          <TableBodyCell>Lido</TableBodyCell>
        </TableRow>
      </TableBody>
    </Table>,
  );

describe('WalletButton', () => {
  it('Renders a table with two columns and one row', () => {
    const { container, getByText } = subject();

    const assetHeaderResult = getByText('Asset');
    const protocolHeaderResult = getByText('Protocol');
    const assetValueResult = getByText('ETH');
    const protocolValueResult = getByText('Lido');

    expect(assetHeaderResult).not.toBeNull();
    expect(protocolHeaderResult).not.toBeNull();
    expect(assetValueResult).not.toBeNull();
    expect(protocolValueResult).not.toBeNull();

    expect(container).toMatchSnapshot();
  });

  it('properly sets text alignment on body cells', () => {
    const { container } = subject();

    const results = container.querySelectorAll('td');

    expect(results.length).toBe(2);

    expect(results[0].dataset.align).toBe('center');
    expect(results[1].dataset.align).toBe('left');
  });

  it('properly sets text alignment on head cells', () => {
    const { container } = subject();

    const results = container.querySelectorAll('th');

    expect(results.length).toBe(2);

    expect(results[0].dataset.align).toBe('right');
    expect(results[1].dataset.align).toBe('left');
  });

  it('row onClick callback is called when user clicks on row', () => {
    const { container } = subject();

    const result = container.querySelector('#first-pool');

    expect(result).not.toBeNull();

    if (result) {
      fireEvent.click(result);
    }

    expect(mockOnClickHandler).toHaveBeenCalledTimes(1);
    expect(mockOnClickHandler).toHaveBeenCalledWith('first-pool');
  });
});
