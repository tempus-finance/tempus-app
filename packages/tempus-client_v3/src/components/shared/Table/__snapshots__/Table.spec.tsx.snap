// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletButton Renders a table with two columns and one row 1`] = `
<div>
  <table
    class="tc__table"
  >
    <thead>
      <tr
        class="tc__tableRow"
        id="header"
      >
        <th
          class="tc__tableHeadCell"
          data-align="right"
        >
          <div
            style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
          >
            Asset
          </div>
        </th>
        <th
          class="tc__tableHeadCell"
          data-align="left"
        >
          <div
            style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
          >
            Protocol
          </div>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="tc__tableRow"
        id="first-pool"
      >
        <td
          class="tc__tableBodyCell"
          data-align="center"
        >
          ETH
        </td>
        <td
          class="tc__tableBodyCell"
          data-align="left"
        >
          Lido
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
